ddterm/shell/extension.js

```diff
@@ -140,6 +142,50 @@ class ExtensionDBusInterface {
     }
 }
 
+class DesktopEntry {
+    constructor() {
+        this.target_file = Gio.File.new_for_path(GLib.build_filenamev(
+            [
+                GLib.get_user_data_dir(),
+                'applications',
+                `${APP_ID}.desktop`,
+            ]
+        ));
+        this.source_file = Gio.File.new_for_path(GLib.build_filenamev(
+            [
+                GLib.get_user_data_dir(),
+                'gnome-shell',
+                'extensions',
+                Me.metadata.uuid,
+                'ddterm',
+                `${APP_ID}.desktop`,
+            ]
+        ));
+    }
+
+    _ensure_install_directory_exists() {
+        var install_directory = this.target_file.get_parent();
+        if (!GLib.file_test(install_directory.get_path(), GLib.FileTest.IS_DIR))
+            install_directory.make_directory_with_parents(null);
```

Comment:
Could you maybe use [`GLib.mkdir_with_parents`](https://gjs-docs.gnome.org/glib20/glib.mkdir_with_parents) that doesn't throw? And thus avoid checking if the directory already exists

Reply:
That's much better indeed. Fixed in [5744f79](https://github.com/ddterm/gnome-shell-extension-ddterm/pull/297/commits/5744f79d92c0ebe210b0365b715bc11da11ce39f).